apps/emqx_resource/src/emqx_resource_manager.erl

```diff
@@ -37,7 +37,6 @@
     list_all/0,
```

Comment:
Just curiosity: `list_all/0` still collects the metrics, and it seems to be used only by `emqx_resource:list_instances_verbose`, which in turn seems to be for debugging purposes.  Maybe we can remove `data_record_to_external_map_with_metrics`?

Reply:
I think the latter would be better, so we could drop the `..._map_with_metrics` fn and use the same API to collect metrics when calling this debug function.  What do you think?

Reply:
I think mostly the same, did that in 9c9f39d0f7431ae7f17b711ff61c17135a223761.